### PR TITLE
Return raw bytes

### PR DESCRIPTION
**Pull request by lpusok**
Return the position of dictionaries in the raw byte stream. They are placed under a custom key in the dictionary.

This allows to overwrite a project with only  the modified parts. Xcode projects contain comments and are sensitive to ordering of objects, this allows to be more compatible with Cordova and the Xcode agvtool tool.

### CODE REVIEW

File: text_parser_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	}{
+		{
+			name: "Read string",
+			parser: textPlistParser{
+				reader: bytes.NewReader([]byte(`"abc"`)),
+			},
+			wantPval: cfString("abc"),
+		},
+		{
+			name: "Map with empy key",
```
Comment (godrei) [inline text_parser_test.go]: typo: `empy` -> `empty`

File: text_parser_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+			parser: textPlistParser{
+				reader: bytes.NewReader([]byte(`{""=Hello;}`)),
+			},
+			wantPval: &cfDictionary{
+				keys:   []string{""},
+				values: []cfValue{cfString("Hello")},
+			},
+		},
+		{
+			name: "Map with empy key",
```
Comment (godrei) [inline text_parser_test.go]: Map with non-empty key

File: go.mod [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
+go 1.15
 
 require (
 	// for cmd/ply
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/kr/pretty v0.1.0 // indirect
+	github.com/stretchr/testify v1.7.0
 	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127 // indirect
-	gopkg.in/yaml.v2 v2.2.1
+	gopkg.in/yaml.v1 v1.0.0-20140924161607-9f9df34309c0
```
Comment (godrei) [inline go.mod:8]: why did we switch to yaml.v1?

File: go.mod [hunk 4]
```diff
@@ -1,9 +1,12 @@
-module howett.net/plist
+module github.com/bitrise-io/go-plist
+
+go 1.15
 
 require (
 	// for cmd/ply
 	github.com/jessevdk/go-flags v1.4.0
 	github.com/kr/pretty v0.1.0 // indirect
+	github.com/stretchr/testify v1.7.0
```
Comment (godrei) [inline go.mod]: NOTE: I'm not sure if it wort introducing this new dependency for 2 calls only

File: decode_test.go [hunk 5]
```diff
@@ -145,11 +161,28 @@ func TestDecode(t *testing.T) {
 					}
 
 					results[fmt] = val
-					if !reflect.DeepEqual(expVal, val) {
-						t.Logf("Expected: %#v\n", expVal)
-						t.Logf("Received: %#v\n", val)
-						t.Fail()
+					require.Equal(t, expVal, val)
+				})
+				subtest(t, FormatNames[fmt], func(t *testing.T) {
```
Comment (godrei) [inline decode_test.go:170]: i do not get this new subtest, it seems the 1. subtests runs the original cases (without TestCustomAnnotiation)
while the 2. subtest runs all the cases, it seems this way we run some test cases 2 times.

Comment (lpusok) [reply decode_test.go:170]: One calls Unmarshal the other the custom Unmarshal

File: go.mod (hunk 3 above)
Comment (lpusok) [reply go.mod:8]: It seems it is referenced in of the files in the ply package.

**Review by godrei — APPROVED** [review]

**Pull request merged**